packages/langium/src/utils/cst-utils.ts

```diff
@@ -81,13 +81,14 @@ export enum RangeComparison {
     After = 1,
     OverlapFront = 2,
     OverlapBack = 3,
-    Inside = 4
+    Inside = 4,
+    Outside = 5,
 }
 
 export function compareRange(range: Range, to: Range): RangeComparison {
-    if (range.end.line < to.start.line || (range.end.line === to.start.line && range.end.character < range.start.character)) {
+    if (range.end.line < to.start.line || (range.end.line === to.start.line && range.end.character <= to.start.character)) {
```

Comment:
There is an IMO unexpected edge case in case the `range` is a zero-width range at the start of `to`. The expected result would be `Inside`, but it currently results in `Before`:

```ts
// This currently results in `Before`
[{ start: { line: 1, character: 1 }, end: { line: 1, character: 1 } }, { start: { line: 1, character: 1 }, end: { line: 1, character: 5 } }]
```

Reply:
> I signed it just now, may need some time for syncing.

Seems to have gone through, thanks!

> Well it's actually expected for me. Zero-width range is a hard-to-imagine concept. I don't think a zero-width range is a real range, it's more like an insertable position.

Yeah, I guess. Especially in the context that we use it, it will probably never appear anyway.